File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    let stack = 0

    for(let i=0; i<s.length; i++) {
     s[i] === "(" ? stack++ : stack--      
        
     if(stack < 0) return false // )가 먼저 나온 경우
    }
    
    return stack === 0 ? true : false // )가 덜 나온 경우 
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    arr = []
    open = "("
    close = ")"
    
    
    for val in s:
        if val == open:
            arr.append(val)
        elif val == close:
            if len(arr) == 0:
                return False
            else: 
                arr.pop()    
            
    if len(arr) != 0:
        return False
    else:
        return True
